pkg/pf/tests/diff_set_test.go

```diff
@@ -0,0 +1,243 @@
+				Optional:    true,
+				ElementType: types.StringType,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.RequiresReplace(),
+				},
+			},
+		},
+	}
+
+	nestedAttributeSchema := rschema.Schema{
+		Attributes: map[string]rschema.Attribute{
+			"key": rschema.SetNestedAttribute{
+				Optional: true,
+				NestedObject: rschema.NestedAttributeObject{
+					Attributes: map[string]rschema.Attribute{
+						"nested": rschema.StringAttribute{Optional: true},
+					},
+				},
+			},
+		},
+	}
+
+	nestedAttributeReplaceSchema := rschema.Schema{
+		Attributes: map[string]rschema.Attribute{
+			"key": rschema.SetNestedAttribute{
+				Optional: true,
+				NestedObject: rschema.NestedAttributeObject{
+					Attributes: map[string]rschema.Attribute{
+						"nested": rschema.StringAttribute{Optional: true},
+					},
+				},
+				PlanModifiers: []planmodifier.Set{
+					setplanmodifier.RequiresReplace(),
+				},
+			},
+		},
+	}
+
+	nestedAttributeNestedReplaceSchema := rschema.Schema{
+		Attributes: map[string]rschema.Attribute{
+			"key": rschema.SetNestedAttribute{
+				Optional: true,
+				NestedObject: rschema.NestedAttributeObject{
+					Attributes: map[string]rschema.Attribute{
+						"nested": rschema.StringAttribute{
+							Optional: true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	blockSchema := rschema.Schema{
+		Blocks: map[string]rschema.Block{
+			"key": rschema.SetNestedBlock{
+				NestedObject: rschema.NestedBlockObject{
+					Attributes: map[string]rschema.Attribute{
+						"nested": rschema.StringAttribute{Optional: true},
+					},
+				},
+			},
+		},
+	}
+
+	blockReplaceSchema := rschema.Schema{
+		Blocks: map[string]rschema.Block{
+			"key": rschema.SetNestedBlock{
+				NestedObject: rschema.NestedBlockObject{
+					Attributes: map[string]rschema.Attribute{
+						"nested": rschema.StringAttribute{
+							Optional: true,
+						},
+					},
+				},
+				PlanModifiers: []planmodifier.Set{
+					setplanmodifier.RequiresReplace(),
+				},
+			},
+		},
+	}
+
+	blockNestedReplaceSchema := rschema.Schema{
+		Blocks: map[string]rschema.Block{
+			"key": rschema.SetNestedBlock{
+				NestedObject: rschema.NestedBlockObject{
+					Attributes: map[string]rschema.Attribute{
+						"nested": rschema.StringAttribute{
+							Optional: true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	attrList := func(arr *[]string) cty.Value {
+		if arr == nil {
+			return cty.NullVal(cty.DynamicPseudoType)
+		}
+		slice := make([]cty.Value, len(*arr))
+		for i, v := range *arr {
+			slice[i] = cty.StringVal(v)
+		}
+		if len(slice) == 0 {
+			return cty.ListValEmpty(cty.String)
+		}
+		return cty.ListVal(slice)
+	}
+
+	nestedAttrList := func(arr *[]string) cty.Value {
+		if arr == nil {
+			return cty.NullVal(cty.DynamicPseudoType)
```

Comment:
I don't understand why `cty.DynamicPseudoType` is the correct type here, but the wrong type for an empty list. Can you explain?